Source/Fuse.Controls.Navigation/NavigationControl.Pages.uno

```diff
@@ -28,28 +27,30 @@ namespace Fuse.Controls
 		int _curPageIndex = -1;
 		void OnPageHistoryChanged()
 		{
-			if (!IsRootingStarted)
+			if (AncestorRouterPage == null || !IsRootingStarted)
 				return;
 				
-			if (_pageHistorySubscription != null)
-			{
-				_pageHistorySubscription.Dispose();
-				_pageHistorySubscription = null;
-			}
+			AncestorRouterPage.ChildRouterPages.Detach();
 			
 			if (_pageHistory == null)
 				return;
-				
+			
 			var obs = _pageHistory as IObservable;
 			if (obs != null)
-				_pageHistorySubscription = obs.Subscribe(this);
+			{
+				AncestorRouterPage.ChildRouterPages.Attach( obs, this );
+			}
+			else
+			{
+				Fuse.Diagnostics.UserError( "PageHistory expects an observable array. It will not work correctly otherwise", this );
+			}
 				
 			_curPageIndex = -1;
 			FullUpdatePages(UpdateFlags.ForceGoto);
+			
+			//TODO: Unroot needs to detach
```

Comment:
Why is this left as a TODO? What goes wrong if it's not fixed?

Reply:
Thanks, I forgot to finish that. It'd a memory leak, amoung other potential weirdness that could happen.